src/ui/public/visualize/loader/pipeline_helpers.js

```diff
@@ -0,0 +1,134 @@
+    pipeline += `tsvb params='${JSON.stringify(visState.params)}'`;
+  } else if (vis.type.name === 'timelion') {
+    pipeline += `timelion_vis expression='${visState.params.expression}' interval='${visState.params.interval}'`;
+  } else if (vis.type.name === 'markdown') {
+    pipeline += `markdown spec='${visState.params.markdown.replace(/'/g, `\\'`)}' 
+    params='${JSON.stringify(visState.params).replace(/'/g, `\\'`)}'`;
+  } else if (vis.type.name === 'table') {
+    const schemas = getSchemas(vis);
+    pipeline += `kibana_table visConfig='${JSON.stringify(visState.params)}' `;
+    if (schemas.split) schemas.split.forEach(split => pipeline += `split='${split}' `);
+    if (schemas.bucket) schemas.bucket.forEach(bucket => pipeline += `bucket='${bucket}' `);
+    schemas.metric.forEach(metric => pipeline += `metric='${metric}' `);
+  } else if (vis.type.name === 'metric') {
+    const schemas = getSchemas(vis);
+    pipeline += `kibana_metric visConfig='${JSON.stringify(visState.params)}' `;
+    if (schemas.bucket) schemas.bucket.forEach(bucket => pipeline += `bucket='${bucket}' `);
+    schemas.metric.forEach(metric => pipeline += `metric='${metric}' `);
+  } else if (vis.type.name === 'tagcloud') {
+    const schemas = getSchemas(vis);
+    pipeline += `tagcloud visConfig='${JSON.stringify(visState.params)}' `;
+    schemas.segment.forEach(bucket => pipeline += `bucket='${bucket}' `);
+    schemas.metric.forEach(metric => pipeline += `metric='${metric}' `);
+  } else if (vis.type.name === 'region_map') {
+    const schemas = getSchemas(vis);
+    pipeline += `regionmap visConfig='${JSON.stringify(visState.params)}' `;
+    schemas.segment.forEach(bucket => pipeline += `bucket='${bucket}' `);
+    schemas.metric.forEach(metric => pipeline += `metric='${metric}' `);
+  } else if (vis.type.name === 'pie') {
+    pipeline += `kibana_pie visConfig='${JSON.stringify(visState.params)}' schemas='${JSON.stringify(getSchemas(vis))}'`;
+  } else if (vis.type.name === 'tile_map') {
+    const schemas = getSchemas(vis);
+    pipeline += `tilemap visConfig='${JSON.stringify(visState.params)}' `;
+    if (schemas.segment) schemas.segment.forEach(bucket => pipeline += `bucket='${bucket}' `);
+    schemas.metric.forEach(metric => pipeline += `metric='${metric}' `);
+  } else if (vislibCharts.includes(vis.type.name)) {
+    pipeline += `vislib type='${vis.type.name}' 
+      visConfig='${JSON.stringify(visState.params)}' schemas='${JSON.stringify(getSchemas(vis))}'`;
+  } else {
+    pipeline += `visualization type='${vis.type.name}' visConfig='${JSON.stringify(visState.params)}'`;
+  }
+
+  return pipeline;
+};
+
+export const runPipeline = async (pipeline, context, handlers) => {
+  try {
+    const ast = fromExpression(pipeline);
+    const pipelineResponse = await interpretAst(ast, context, handlers);
+    return pipelineResponse;
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log(e, pipeline);
```

Comment:
I think we want to throw this as an error, so we actually show a toast in the infrastructure, to the user.